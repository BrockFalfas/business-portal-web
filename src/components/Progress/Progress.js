import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Progress as AntProgress } from 'antd';

import './Progress.scss';

export class Progress extends React.Component {
  static propTypes = {
    percent: PropTypes.number,
  };

  static defaultProps = {
    percent: 0,
  };

  render() {
    const { percent, ...progressProps } = this.props;
    return (
      <div className={classnames('Progress', { 'Progress--success': percent === 100 })}>
        <AntProgress {...progressProps} showInfo={false} percent={percent} />
        <div className="Progress-text" style={{ left: `${percent / 2}%` }}>
          {percent} %
        </div>
      </div>
    );
  }
}

export default Progress;
