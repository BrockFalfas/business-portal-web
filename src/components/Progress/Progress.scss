@import "../../styles/var";

.Progress {
  position: relative;

  &-text {
    position: absolute;
    bottom: calc(100% - 5px);
    transform: translateX(-50%);
    color: $progress-default-color;
  }

  &--success &-text {
    color: $success-color;
  }
}
