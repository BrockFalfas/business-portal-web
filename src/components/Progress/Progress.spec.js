import React from 'react';
import { shallow } from 'enzyme';

import Progress from './Progress';

describe('component: <Progress/>', () => {
  describe('Progress', () => {
    it('should render without crashing', () => {
      const wrapper = shallow(<Progress />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should pass all the other props to the Ant Progress component', () => {
      const wrapper = shallow(<Progress percent={20} status="exception" />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render the percent text in the middle', () => {
      const wrapper = shallow(<Progress percent={20} status="exception" />);
      expect(wrapper.find('.Progress-text').prop('style')).toEqual({ left: '10%' });
    });
  });
});
