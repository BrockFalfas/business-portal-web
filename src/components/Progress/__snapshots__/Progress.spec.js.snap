// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <Progress/> Progress should pass all the other props to the Ant Progress component 1`] = `
<div
  className="Progress"
>
  <Progress
    percent={20}
    showInfo={false}
    status="exception"
  />
  <div
    className="Progress-text"
    style={
      Object {
        "left": "10%",
      }
    }
  >
    20
     %
  </div>
</div>
`;

exports[`component: <Progress/> Progress should render without crashing 1`] = `
<div
  className="Progress"
>
  <Progress
    percent={0}
    showInfo={false}
  />
  <div
    className="Progress-text"
    style={
      Object {
        "left": "0%",
      }
    }
  >
    0
     %
  </div>
</div>
`;
